src/services/MethodService.ts

```diff
@@ -0,0 +1,127 @@
+            .forEach((z) => {
+                this.writer.withIndentationLevel(3, () => this.writer.writeLine(`${z.name},`));
+            });
+
+        if (this.needPipe(this.model.returnType)) {
+            this.writer.writeLine(`).pipe(${this.createPipe(this.model.returnType)});`);
+            return;
+        }
+
+        this.writer.writeLine(');');
+    }
+
+    private needPipe(returnType: IReturnType | undefined): returnType is IReturnType {
+        if (!returnType) {
+            return false;
+        }
+
+        return [PropertyKind.Object, PropertyKind.Identity, PropertyKind.Guid, PropertyKind.Date].includes(returnType.type.kind);
+    }
+
+    private createPipe(returnType: IReturnType): string {
+        if (returnType.type.kind === PropertyKind.Guid) {
+            return `${this.mappersNamespace}.mapGuid()`;
+        }
+
+        if (returnType.type.kind === PropertyKind.Date) {
+            return `${this.mappersNamespace}.mapDate()`;
+        }
+
+        const type = `${this.mappersNamespace}.${returnType.type.type}`;
+        if (returnType.isCollection) {
+            if (returnType.type.kind === PropertyKind.Identity) {
+                return `${this.mappersNamespace}.mapIdentityCollection(${type})`;
+            }
+
+            return `${this.mappersNamespace}.mapCollection(${type})`;
+        }
+
+        if (returnType.type.kind === PropertyKind.Identity) {
+            return `${this.mappersNamespace}.mapIdentitySingle(${type})`;
+        }
+
+        return `${this.mappersNamespace}.mapSingle(${type})`;
+    }
+
+    private getReturnTypeName(returnType: IReturnType | undefined, targetType: string | undefined): string {
+        if (!returnType || !targetType) {
+            return 'void';
+        }
+
+        return this.getFullTypeName(targetType, returnType.isCollection, returnType.isModel);
+    }
+
+    private getFullTypeName(type: string, isCollection: boolean, isModel: boolean): string {
+        const arraySymbol = isCollection ? '[]' : '';
+        return `${isModel ? `${this.modelsNamespace}.` : ''}${type}${arraySymbol}`;
+    }
+
+    public getFullMethodName(model: IMethodModel): string {
+        const queryParams = this.getQueryParams(model.parameters);
+        const pathParam = this.getPathParam(model.parameters);
+        const path = '`' + `${model.name}${pathParam}${queryParams}` + '`'
+        return path;
+    }
+
+    private getQueryParams(params: IMethodParameterModel[]): string {
+        const pairs = params
+            .filter((z) => z.place === ParameterPlace.Query)
+            .map((z) => `${z.name}=\${encodeURIComponent(${z.name})}`);
+
+        if (!pairs.length) {
+            return '';
+        }
+
+        return `?${pairs.join('&')}`;
+    }
+
+    private getPathParam(params: IMethodParameterModel[]): string {
+        const path = params
+            .find((z) => z.place === ParameterPlace.Path)?.name;
+
+        if (!path) {
+            return '';
+        }
```

Comment:
what will be if endpoint has multiple path params.
For example /api/product/56/details/89

Reply:
I use originUri for building correct method name with multiple path params